multipledispatch/utils.py

```diff
@@ -45,18 +48,19 @@ def _toposort(edges):
     [2] http://en.wikipedia.org/wiki/Toposort#Algorithms
     """
     incoming_edges = reverse_dict(edges)
-    incoming_edges = dict((k, set(val)) for k, val in incoming_edges.items())
-    S = set((v for v in edges if v not in incoming_edges))
+    incoming_edges = OrderedDict((k, set(val))
+                                 for k, val in incoming_edges.items())
+    S = OrderedDict.fromkeys((v for v in edges if v not in incoming_edges))
```

Comment:
@Andrew-peng we can remove one level of parentheses here.

Reply:
👍